File: Programmers/Lv_1/두개뽑아서더하기.js
function solution(numbers) {
    let temp = [];

    for (let i = 0; i < numbers.length-1; i++) {
        for (let j = i+1; j < numbers.length; j++) {
            temp.push(numbers[i] + numbers[j]);
        }
    }

    const answer = [...new Set(temp)];

    return answer.sort((a, b) => a-b);
}

function solution2(numbers) {
    var answer = [];

    for (let i = 0; i < numbers.length; i++) {
        for (let j = 0; j < numbers.length; j++) {
            let sum = numbers[i] + numbers[j];
            
            if (answer.indexOf(sum) === -1) answer.push(sum);
        }
    }

    answer.sort((a, b) => a-b);
}

File: Programmers/Lv_1/두개뽑아서더하기.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;

    for (int i = 0; i < numbers.size()-1; i++) {
        for (int j = i+1; j < numbers.size(); j++) {
            answer.push_back(numbers[i] + numbers[j]);
        }
    }

    sort(answer.begin(), answer.end());
    answer.erase(unique(answer.begin(), answer.end()), answer.end());

    return answer;
}

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int n, data;
    vector<int> numbers, result;
    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> data;
        numbers.push_back(data);
    }

    result = solution(numbers);

    for (int i = 0; i < result.size(); i++) {
        cout << result[i] << " ";
    }

    return 0;
}